### PR TITLE
Separate installation of repos

### PR DESCRIPTION
**Pull request by pyrodie18**
##### SUMMARY
This PR will remove the repo installation tasks from the individual roles and consolidate them into a single role that each of the roles references as requested in #1242.

Also adds the downloading of the GPG key for RHEL based operating systems as was started in https://github.com/ansible-collections/community.zabbix/pull/1239 (credit to @jonathananspw)

All adds support for yum authentication on RHEL based operating systems as was started in https://github.com/ansible-collections/community.zabbix/pull/1185 (credit to @user78012)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
All roles

### CODE REVIEW

File: docs/ZABBIX_MANAGE_REPO_ROLE.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * Ubuntu
+
+So, you'll need one of those operating systems.. :-)
+Please send Pull Requests or suggestions when you want to use this role for other Operating systems.
+
+## Zabbix Versions
+
+See the following list of supported Operating systems with the Zabbix releases:
+
+| Zabbix              | 6.4 | 6.2 | 6.0 |
```
Comment (BGmot) [inline docs/ZABBIX_MANAGE_REPO_ROLE.md:38]: Remove 6.2? Add 7.0?

File: molecule/zabbix_manage_repo/molecule.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ANSIBLE_ROLES_PATH: $HOME/.ansible/collections/ansible_collections/community/zabbix/roles
+  inventory:
+    group_vars:
+      all:
+        zabbix_repo_apt_priority: 1
+        zabbix_repo_yum_gpg_check: 1
+      v64:
+        zabbix_server_version: 6.4
+      v62:
+        zabbix_server_version: 6.2
```
Comment (BGmot) [inline molecule/zabbix_manage_repo/molecule.yml:30]: Remove 6.2? Add 7.0?

File: roles/zabbix_agent/vars/Debian.yml [hunk 3]
```diff
@@ -33,6 +33,3 @@ zabbix_valid_agent_versions:
     - 6.4
     - 6.2
```
Comment (BGmot) [inline roles/zabbix_agent/vars/Debian.yml:34]: Remove 6.2? add 7.0?

File: roles/zabbix_manage_repo/requirements.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+- src: geerlingguy.apache
+- src: geerlingguy.mysql
+- src: geerlingguy.postgresql
+- src: community.postgresql
```
Comment (BGmot) [inline roles/zabbix_manage_repo/requirements.yml:5]: Sorry, why do we need these `- src:` ?

File: roles/zabbix_web/vars/Debian.yml [hunk 5]
```diff
@@ -44,6 +44,3 @@ zabbix_valid_web_versions:
     - 6.4
     - 6.2
```
Comment (BGmot) [inline roles/zabbix_web/vars/Debian.yml:45]: Remove 6.2? add 7.0?

File: docs/ZABBIX_MANAGE_REPO_ROLE.md (hunk 1 above)
Comment (pyrodie18) [reply docs/ZABBIX_MANAGE_REPO_ROLE.md:38]: Didn't add 7.0 yet (doing things one at a time).  But removing 6.2

File: molecule/zabbix_manage_repo/molecule.yml (hunk 2 above)
Comment (pyrodie18) [reply molecule/zabbix_manage_repo/molecule.yml:30]: Didn't add 7.0 yet (doing things one at a time).  But removing 6.2

File: roles/zabbix_manage_repo/requirements.yml (hunk 4 above)
Comment (pyrodie18) [reply roles/zabbix_manage_repo/requirements.yml:5]: Never noticed it....It was actually in the server role which is what I copied when I created this.  Removing from both of them.

File: roles/zabbix_web/vars/Debian.yml (hunk 5 above)
Comment (pyrodie18) [reply roles/zabbix_web/vars/Debian.yml:45]: Didn't add 7.0 yet (doing things one at a time).  But removing 6.2

**Comment by eb4x** [comment]
Would you mind changing the role name to just `zabbix_repo` so it matches what the prefixes are (or will be) in `defaults/main.yml`?

It just keeps things (mostly) consistent between all role names and their defaults.

File: roles/zabbix_proxy/tasks/Debian.yml [hunk 6]
```diff
@@ -1,103 +1,7 @@
...
-    zabbix_underscore_version: "{{ zabbix_proxy_version | regex_replace('\\.', '_') }}"
-  tags:
-    - always
-
 - name: "Debian | Set some facts for Zabbix"
   ansible.builtin.set_fact:
     datafiles_path: /usr/share/doc/zabbix-sql-scripts/{{ zabbix_proxy_db_long }}
   tags:
     - install
     - config
```
Comment (eb4x) [inline roles/zabbix_proxy/tasks/Debian.yml:7]: You can kill off this code aswell, it's zabbix-5 related and goes completely unused.

File: roles/zabbix_server/tasks/RedHat.yml [hunk 7]
```diff
@@ -1,39 +1,6 @@
...
-  register: yum_repo_installed
-  become: true
-  notify:
-    - "clean repo files from proxy creds"
-  tags:
-    - install
-
 - name: "RedHat | Configure SELinux when enabled"
   ansible.builtin.include_tasks: selinux.yml
   when: ansible_facts.selinux.status | default('disabled') == 'enabled'
```
Comment (eb4x) [inline roles/zabbix_server/tasks/RedHat.yml:6]: Same here, just move this to main.yml and we can drop the Debian/RedHat tasks completely :smile: !

File: roles/zabbix_agent/tasks/main.yml [hunk 8]
```diff
@@ -87,6 +87,14 @@
     zabbix_api_server_port: "{{ '443' if zabbix_api_use_ssl|bool else '80' }}"
   when: zabbix_api_server_port is undefined
 
+- name: Install Repository
+  ansible.builtin.include_role:
+    name: community.zabbix.zabbix_manage_repo
```
Comment (eb4x) [inline roles/zabbix_agent/tasks/main.yml:92]: Just an additional thought going with renaming this role to just `zabbix_repo`, have a `when: zabbix_manage_repo | default(true)`  condition, allowing people to not run this code, incase they have their own repo-mirror-setup-thing they need to use.

**Comment by pyrodie18** [comment]
> Would you mind changing the role name to just `zabbix_repo` so it matches what the prefixes are (or will be) in `defaults/main.yml`?
> 
> It just keeps things (mostly) consistent between all role names and their defaults.

To easy

File: roles/zabbix_proxy/tasks/Debian.yml (hunk 6 above)
Comment (pyrodie18) [reply roles/zabbix_proxy/tasks/Debian.yml:7]: done

File: roles/zabbix_proxy/tasks/RedHat.yml [hunk 9]
```diff
@@ -1,39 +1,6 @@
...
-    proxy: "{{ zabbix_http_proxy | default(omit) }}"
-  with_items: "{{ zabbix_repo_yum }}"
-  register: yum_repo_installed
-  become: true
-  notify:
-    - "clean repo files from proxy creds"
-  tags:
-    - install
-
 - name: "Configure SELinux when enabled"
```
[the remark this answers is not in the document]
Comment (pyrodie18) [reply roles/zabbix_proxy/tasks/RedHat.yml:4]: done

File: roles/zabbix_server/tasks/RedHat.yml (hunk 7 above)
Comment (pyrodie18) [reply roles/zabbix_server/tasks/RedHat.yml:6]: done

File: roles/zabbix_proxy/tasks/main.yml [hunk 10]
```diff
@@ -46,8 +45,20 @@
...
+    zabbix_repo_apt_priority: "{{ zabbix_proxy_apt_priority | default (omit) }}"
+    zabbix_repo_version: "{{ zabbix_proxy_version }}"
+    zabbix_repo_package: "zabbix-proxy-{{ zabbix_proxy_database }}"
+  when: zabbix_manage_repo | default(true)
+
+- name: "Configure SELinux when enabled"
+  ansible.builtin.include_tasks: selinux.yml
+  when:
+    - zabbix_proxy_selinux | default (false) | bool
+    - ansiblle_os_family == "RedHat"
```
Comment (eb4x) [inline roles/zabbix_proxy/tasks/main.yml:61]: spelling error here.

File: roles/zabbix_server/tasks/main.yml [hunk 11]
```diff
@@ -27,8 +27,20 @@
...
+  vars:
+    zabbix_repo_apt_priority: "{{ zabbix_server_apt_priority | default (omit) }}"
+    zabbix_repo_version: "{{ zabbix_server_version }}"
+    zabbix_repo_package: "zabbix_server-{{ zabbix_proxy_database }}"
+  when: zabbix_manage_repo | default(true)
+
+- name: "Configure SELinux when enabled"
+  ansible.builtin.include_tasks: selinux.yml
+  when:
+    - zabbix_server_selinux | default (false) | bool
```
Comment (eb4x) [inline roles/zabbix_server/tasks/main.yml:42]: You sure we don't want to reuse the previous `when: ansible_facts.selinux.status | default('disabled') == 'enabled'` here? i.e. anything that reports selinux_enabled we apply on. (Suse also comes with selinux, #1194 and I think he would have an easier job (smaller PR) adding support after this merges.)

Comment (eb4x) [reply roles/zabbix_server/tasks/main.yml:42]: Hmm, actually I think even better would be to put this when condition into defaults, so users can explicitly override us. Shame I didn't think about that in the selinux refactor.

```
--- # defaults/main.yml
zabbix_server_selinux: "{{ ansible_facts.selinux.status | default('disabled') == 'enabled' }}"
```

Comment (pyrodie18) [reply roles/zabbix_server/tasks/main.yml:42]: Don't remember where we had that but now that you say it ya I remember seeing that.  Ya I think I like that better.

**Comment by pyrodie18** [comment]
@BGmot  I think this one is ready to go if you want to take a peak

File: roles/zabbix_javagateway/vars/Debian.yml [hunk 12]
```diff
@@ -20,6 +20,3 @@ zabbix_valid_javagateway_versions:
     - 6.4
     - 6.2
```
Comment (BGmot) [inline roles/zabbix_javagateway/vars/Debian.yml:21]: Please remove 6.2

File: roles/zabbix_server/vars/Debian.yml [hunk 13]
```diff
@@ -27,9 +27,6 @@ zabbix_valid_server_versions:
     - 6.2
```
Comment (BGmot) [inline roles/zabbix_server/vars/Debian.yml:27]: Please remove 6.2

File: roles/zabbix_web/vars/Debian.yml (hunk 5 above)
Comment (BGmot) [reply roles/zabbix_web/vars/Debian.yml:45]: I am still seeing 6.2 ? or ...?

**Review by BGmot — CHANGES_REQUESTED** [review]
Confused... still seeing 6.2 in some places. Have you removed them?

**Comment by pyrodie18** [comment]
Ya I removed 6.2 in a different PR..... #1274 I think is ready so if you want to approve that one too it should automagically take care of this one.

**Review by BGmot — APPROVED** [review]
Great job! Thanks!

**Pull request merged**